halo/evmengine/keeper/abci_internal_test.go

```diff
@@ -126,13 +126,14 @@ func TestKeeper_PrepareProposal(t *testing.T) {
 		for _, tt := range tests {
 			t.Run(tt.name, func(t *testing.T) {
 				t.Parallel()
-				ctx, storeService := setupCtxStore(t)
+				ctx, storeService := setupCtxStore(t, cmtproto.Header{Time: cmttime.Now()})
```

Comment:
can reduce boilerplate by allowing to pass `nil` for header in which case `setupCtx` will just do `cmtproto.Header{Time: cmttime.Now()}`. Better is to use functional options.